core/src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -257,8 +286,8 @@ public void postParse(ParseContext context) throws IOException {
     @Override
     protected void parseCreateField(ParseContext context, List<Field> fields) throws IOException {
         boolean parent = context.docMapper().isParent(context.type());
-        if (parent && fieldType().hasDocValues()) {
-            fields.add(createJoinField(context.type(), context.id()));
+        if (parent && joinFieldTypeRef.get().hasDocValues()) {
+            fields.add(new SortedDocValuesField(joinField(context.type()), new BytesRef(context.id())));
```

Comment:
If a document is both parent and child then it has two join fields. So maybe we should have two `MappedFieldType`?